dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java

```diff
@@ -648,9 +648,14 @@ public Result queryProcessDefinitionListPaging(User loginUser, long projectCode,
             }
         }
 
-        int delete = processDefinitionMapper.deleteById(processDefinition.getId());
+        int deleteProcessDefinition = processDefinitionMapper.deleteById(processDefinition.getId());
+        List<ProcessTaskRelation> processTaskRelations = processTaskRelationMapper.queryByProcessCode(project.getCode(), processDefinition.getCode());
+        int deleteRelatedTask = 1;
+        for (ProcessTaskRelation processTaskRelation : processTaskRelations) {
+            deleteRelatedTask &= taskDefinitionMapper.deleteByCode(processTaskRelation.getPostTaskCode());
```

Comment:
Thanks for your contribution. It can't be modified here. The initial design is that tasks are organized through workflows. Tasks and workflows are parallel. Subsequent workflows will support task references and have an interface to operate tasks separately